org.adempiere.base/src/org/compiere/model/MDocumentStatus.java

```diff
@@ -59,16 +59,33 @@ public MDocumentStatus(Properties ctx, ResultSet rs, String trxName) {
 	 *	@return array of document status
 	 */
 	public static MDocumentStatus[] getDocumentStatusIndicators(Properties ctx, int AD_User_ID, int AD_Role_ID)
+	{
+		return getDocumentStatusIndicators(ctx,AD_User_ID,AD_Role_ID);
+	}
+	
+	/**
+	 * Get Document Status Indicators
+	 * @param ctx
+	 * @param AD_User_ID
+	 * @param AD_Role_ID
+	 * @param trxName
+	 * @return array of document status
+	 */
+	public static MDocumentStatus[] getDocumentStatusIndicators(Properties ctx, int AD_User_ID, int AD_Role_ID, String trxName)
 	{
 		if (AD_User_ID < 0)
 			return new MDocumentStatus[0];
 
-		String whereClause = "AD_Client_ID IN (0,?) AND ((AD_User_ID IS NULL OR AD_User_ID=?) AND ( AD_Role_ID IS NULL OR AD_Role_ID=?))";
+		String whereClause = "PA_DocumentStatus.AD_Client_ID IN (0,?) AND ((dsa.AD_User_ID IS NULL OR dsa.AD_User_ID=?) "
+				+ "AND ( dsa.AD_Role_ID IS NULL OR dsa.AD_Role_ID=?) AND (dsa.AD_Client_ID IS NULL OR dsa.AD_Client_ID IN (0,?)))";
+		String joinClause = "LEFT JOIN PA_DocumentStatusAccess dsa ON PA_DocumentStatus.PA_DocumentStatus_ID = dsa.PA_DocumentStatus_ID "
+				+ "AND dsa.IsActive = 'Y' ";
```

Comment:
the dsa.AD_Client_ID condition should be moved to the LEFT JOIN, i.e AND dsa.AD_Client_ID IN (0, PA_DocumentStatus.AD_Client_ID). Perhaps it should be just AND PA_DocumentStatus.AD_Client_ID=dsa.AD_Client_ID instead. I'm curious how dsa at System level (AD_Client_ID=0) can work since Role(AD_Role_ID) is at tenant level (unless you are talking about master and included role but this SQL clause doesn't cater for that either).